packages/edit-site/src/components/layout/router.js

```diff
@@ -64,14 +80,24 @@ export default function useLayoutAreas() {
 
 	// Regular other post types
 	if ( postType && postId ) {
+		let sidebar;
+		if ( postType === 'wp_template_part' || postType === 'wp_block' ) {
+			sidebar = <SidebarNavigationScreenPattern />;
+		} else if ( postType === 'wp_template' ) {
+			sidebar = <SidebarNavigationScreenTemplate />;
+		} else if ( postType === 'page' ) {
+			sidebar = <SidebarNavigationScreenPage />;
+		} else {
+			sidebar = <SidebarNavigationScreenNavigationMenu />;
+		}
 		return {
 			key: 'page',
 			areas: {
+				sidebar,
 				preview: <Editor isLoading={ isSiteEditorLoading } />,
-				mobile:
-					canvas === 'edit' ? (
-						<Editor isLoading={ isSiteEditorLoading } />
-					) : undefined,
+				mobile: canvas === 'edit' && (
```

Comment:
This refactoring made things a lot more simple to follow, thanks @jsnajdr!

I find one little thing missing: fully controlling the mobile views. At the moment, we only use `areas.mobile` for displaying components other than the sidebar. The logic to default to the sidebar if `areas.mobile` is undefined still lives in the layout. Absorbing as part of the router would simplify things further.